app/controllers/application_controller.rb

```diff
@@ -12,8 +12,36 @@ class ApplicationController < ActionController::Base
 
   layout proc { kind_of?(Europeana::Styleguide) ? false : 'application' }
 
-  rescue_from ActiveRecord::RecordNotFound, ActionController::RoutingError do
-    render_error_page(404)
+  rescue_from StandardError do |exception|
+    render_error_page(exception, 500)
+  end
+
+  rescue_from ActiveRecord::RecordNotFound, ActionController::RoutingError do |exception|
+    render_error_page(exception, 404)
+  end
+
+  rescue_from Europeana::API::Errors::RequestError do |exception|
+    if exception.message.match(/Invalid record identifier/)
+      render_error_page(exception, 404)
+    else
+      raise
+    end
+  end
+
+  rescue_from CanCan::AccessDenied do |exception|
+    render_error_page(exception, 403)
+  end
+
+  rescue_from ActionController::UnknownFormat do |exception|
+    render_error_page(exception, 500, 'html')
+  end
+
+  def log_error(exception)
+    trace = Rails.backtrace_cleaner.clean(exception.backtrace)
+    message = "\n#{exception.class} (#{exception.message}):\n"
+    message << exception.annoted_source_code.to_s if exception.respond_to?(:annoted_source_code)
+    message << "  " << trace.join("\n  ")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.